lms/djangoapps/courseware/access.py

```diff
@@ -136,8 +136,11 @@ def has_access(user, action, obj, course_key=None):
     if not user:
         user = AnonymousUser()
 
-    if in_preview_mode():
-        if not bool(has_staff_access_to_preview_mode(user=user, obj=obj, course_key=course_key)):
+    # Only global staff and course team should have access to preview mode.
+    # User, masquerading as student is allowed, as it is staff which can use masquerading.
+    if in_preview_mode() and course_key:
+        if not (any(administrative_accesses_to_course_for_user(user, course_key)) or
```

Comment:
can you refactor `has_staff_access_to_preview_mode` to be:

```python
def has_staff_access_to_preview_mode(user, course_key):
    has_admin_access_to_course = any(administrative_accesses_to_course_for_user(user, course_key)
    if has_admin_access_to_course or is_masquerading_as_student(user, course_key):
        return ACCESS_DENIED
    return ACCESS_GRANTED
```
That way, this part of the code remains semantically the same, but the implementation of `has_staff_access_to_preview_mode` changes